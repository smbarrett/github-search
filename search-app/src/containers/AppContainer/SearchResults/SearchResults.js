import React, { Component } from 'react';
import Table from '@mui/material/Table';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';

class SearchResults extends Component {
    constructor(props) {
        super(props);
    }

    renderRow(result) {
        const { name, owner, stars, url } = result;

        return (
            <TableRow key={`${name}-${owner}`}>
                <TableCell>
                    <a href={url}>{name}</a>
                </TableCell>
                <TableCell>{owner}</TableCell>
                <TableCell>{stars}</TableCell>
            </TableRow>
        );
    }

    render() {
        const { results } = this.props; 

        if (results && !results.length) {
            return (
                <div>No Results Found</div>
            );
        }

        return (
            <div>
                <TableContainer component={Paper}>
                    <Table>
                        <TableHead>
                            <TableRow>
                                <TableCell>Name</TableCell>
                                <TableCell>Owner</TableCell>
                                <TableCell>Stars</TableCell>
                            </TableRow>
                            {results.map(result => this.renderRow(result))}
                        </TableHead>
                    </Table>
                </TableContainer>
            </div>
        );
    }
}

export default SearchResults;