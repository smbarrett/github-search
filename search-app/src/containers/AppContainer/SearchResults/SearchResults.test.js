import React from 'react';
import renderer from 'react-test-renderer';
import SearchResults from './SearchResults';

test('Render SearchResults', () => {
    let results = [
        {
            name: 'repoName',
            owner: 'owner',
            stars: 10,
            url: 'urlToRepo',
        }
    ];
    const component = renderer.create(
        <SearchResults results={results}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
