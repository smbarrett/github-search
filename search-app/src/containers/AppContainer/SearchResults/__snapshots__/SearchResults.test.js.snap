// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SearchResults 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root css-rqglhn-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Owner
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Stars
          </th>
        </tr>
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            <a
              href="urlToRepo"
            >
              repoName
            </a>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            owner
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            10
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
